src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +87,114 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        return this.requestRemoteDataAndMerge(this.schema, schema).then(action(() => {
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.validate();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    requestRemoteDataAndMerge = (localSchema: Schema, remoteSchema: Schema) => {
+        if (localSchema) {
+            return this.resourceStore.requestRemoteData({template: this.type}).then((data: Object) => {
+                const result = this.mergeData(localSchema, remoteSchema, this.data, data);
+                this.setMultiple(result);
+            });
+        }
+        return Promise.resolve();
+    };
+
+    mergeData(
+        localSchema: Schema,
+        remoteSchema: Schema,
+        localData: Object,
+        remoteData: Object
+    ) {
+        let result = {};
+        if (!localSchema || !remoteSchema) {
+            return result;
+        }
+
+        for (const name in remoteSchema) {
+            const {items: remoteItems, type: remoteType, types: remoteTypes} = remoteSchema[name];
+            const {items: localItems, type: localType, types: localTypes} = localSchema[name] || {};
+
+            if (remoteType === SECTION_TYPE && remoteItems &&
+                localType === SECTION_TYPE && localItems) {
+                result = this.mergeData(
+                    localItems,
+                    remoteItems,
+                    localData,
+                    remoteData
+                );
+                continue;
+            }
+            if (remoteTypes && localTypes
+                && Object.keys(remoteTypes).length > 0 && Object.keys(localTypes).length > 0
+                && localData[name] && remoteData[name]
+                && isArrayLike(localData[name]) && isArrayLike(remoteData[name])
+            ) {
+                for (const key of remoteData[name].keys()) {
+                    const remoteChildData = remoteData[name][key];
+                    const localChildData = toJS(localData[name].length > key ? localData[name][key] || {} : {});
+
+                    const localChildSchema = localTypes[localChildData.type]?.form;
+                    const remoteChildSchema = remoteTypes[remoteChildData.type].form;
+
+                    if (!(localChildData.type in remoteTypes)) {
```

Comment:
do we need to do this befor the merge? think i would move it below the merge and change the `resultChildData` otherwise

Reply:
guess we need to move it into the `else` above `result[name][key] = resultChildData;` because of the `Object.keys(resultChildData).length` check